File: solutions/Javascript/Score of Parentheses.js
var scoreOfParentheses = function (s) {
  let stack = [];
  let res = 0;
  let i = 0;
  while (i < s.length) {
    if (s[i] === '(') {
      stack.push(s[i]);
    } else {
      if (stack[stack.length - 1] == '(') {
        stack.pop();
        stack.push(1);
      } else {
        let inner_count = 0;
        while (stack[stack.length - 1] !== '(') {
          inner_count += stack.pop();
        }
        stack.pop();
        stack.push(2 * inner_count);
      }
    }
    i += 1;
    console.log(stack);
  }
  while (stack.length !== 0) {
    res += stack.pop();
  }
  return res;
};

File: solutions/python/Score of Parentheses.py
class Solution:
    def scoreOfParentheses(self, s: str) -> int:
        stack=[]
        i=0
        res=0
        while i<len(s):
            if s[i] == '(':
                stack.append(s[i]);
            else:
                if(stack[-1]=='('):
                    stack.pop();
                    stack.append('1')
                else:
                    count=0;
                    while stack[-1]!='(':
                        count+=int(stack.pop())
                    stack.pop();
                    stack.append(str(2*count));
            print(stack)
            i=i+1;
        print(stack);
        while len(stack):
            res+=int(stack.pop())
        return res;